PypeS/pypescript.py

```diff
@@ -598,17 +607,25 @@ def IORead(self):
     def IOWrite(self):
         for member in self.OutputMembers:
             if member.MemberIO:
-                exec('filename = self.' + self.GetIOOutputFileNameMember(member.MemberName))
-                exec('input = self.' + member.MemberName)
+                filename = eval('self.' + self.GetIOOutputFileNameMember(member.MemberName))
+                writerInput = eval('self.' + member.MemberName)
                 if filename:
+                    # Create code object representing local import of requested IO script
                     try:
-                        exec('import ' + member.MemberIO)
+                        writerModule = importlib.import_module('vmtk.'+member.MemberIO)
+                        # Find the principle class to instantiate the requested action defined inside the requested writerModule script.
+                        # Returns a single member list (containing the principle class name) which satisfies the following criteria:
+                        #   1) is a class defined within the script
+                        #   2) the class contains an attribute named 'self.SetScriptName'
+                        # This makes the assumption that only one class per vmtkscript file contains the self.SetScriptName attribute, and that
+                        # the class containing this attribute is the primary class to instantiate a new call to the scripts functionality
+                        writerModuleClasses = [x for x in dir(writerModule) if isclass(getattr(writerModule, x)) and hasattr(getattr(writerModule, x), 'SetScriptName')]
```

Comment:
Same as above. I'm fine with checking the method name too, I'd just like to know why you opted for this solution vs checking the parent class.